### PR TITLE
Fixed an issue where **validate** falsly failed with error `DS108` on empty descriptions.

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes:

## Description
Fixed an issue where **validate** falsly failed with error `DS108` on empty descriptions.

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 1]
```diff
@@ -793,16 +793,27 @@ def is_line_ends_with_dot(self, dict_to_test: dict, arg_field: str):
...
         for output in dict_to_test.get("outputs") or []:
             stripped_description = strip_description(output.get("description", ""))
-            if not stripped_description.endswith(".") and not is_string_ends_with_url(
-                stripped_description
-            ):
+            if self.is_valid_description(stripped_description):
                 line_with_missing_dot += f"The context path {output.get('contextPath')} description should end with a period.\n"
         return line_with_missing_dot
+
+    def is_valid_description(self, stripped_description):
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:804]:
```suggestion
    def is_valid_description(self, stripped_description: str) -> bool:
```

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 2]
```diff
@@ -793,16 +793,29 @@ def is_line_ends_with_dot(self, dict_to_test: dict, arg_field: str):
...
+               The description string ends with an URL.
+            OR
+               The description string is empty
+        """
+        return (
+            stripped_description
+            and not stripped_description.endswith(".")
+            and not is_string_ends_with_url(stripped_description)
+            and not len(stripped_description) == 0
+        )
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:821]:
```suggestion
        return (
            not stripped_description.endswith(".")
            and not is_string_ends_with_url(stripped_description)
            and not len(stripped_description) == 0
        )
```

isn't this enough? Its already a string so if `stripped_description` evaluates to `True` it means its already not empty

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 3]
```diff
@@ -793,16 +793,29 @@ def is_line_ends_with_dot(self, dict_to_test: dict, arg_field: str):
...
+        """
+        Args:
+            stripped_description: (str) a description or comment section from script / integration yml.
+        Verifies the following::
+            1. The description string exists (was extracted correctly from its field).
+            2. The description string ends with a dot.
+            OR
+               The description string ends with an URL.
+            OR
+               The description string is empty
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:814]:
```suggestion
        Verifies the following:
            1. The description string is not a URL
            2. The description string does not end with a dot.
            3. The description string is not empty
            
        Args:
            stripped_description: (str) a description or comment section from script / integration yml.
       ```

**Review by GuyAfik — COMMENTED** [review]
@YuvHayun champ, see my comments

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 3 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:814]: It checks for either one of these conditions to hold, so I think leaving the 'OR's is more informational @GuyAfik

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 2 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:821]: good point

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py (hunk 3 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py]: but you have AND between them and not OR, so what am i missing here?

Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py]: in your docstring it says the opposite, verifies that the string ends with an URL or that the string is empty,
but you actually validate the opposite of what you write in the doscstring, so it does not make sense to me

Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py]: I changed to OR but I'll rephrase it

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 4]
```diff
@@ -793,16 +793,25 @@ def is_line_ends_with_dot(self, dict_to_test: dict, arg_field: str):
...
+        Return False if none of the following conditions are met:
+        - The description string doesn't exist/empty.
+        - The description string ends with a dot.
+        - The description string ends with an URL.
+        """
+        return not (
+            not stripped_description
+            or stripped_description.endswith(".")
+            or is_string_ends_with_url(stripped_description)
+        )
```
Comment (YuvHayun) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:817]: @GuyAfik

Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:817]: I think the previous implementation is better, more readable IMO

**Review by GuyAfik — APPROVED** [review]
see my comment, and then merge 👑

**Pull request merged**